presto-main/src/main/java/io/prestosql/operator/ExchangeOperator.java

```diff
@@ -92,6 +92,7 @@ public void noMoreOperators()
     private final PlanNodeId sourceId;
```

Comment:
Change commit name `Cache ExchangeClient#isBlocked` future. Add explanation why `io.prestosql.operator.ExchangeClient#isBlocked` creates a new future every time (because they can be cancelled in `io.prestosql.server.protocol.Query#waitForResults`)